### ISSUE TITLE
[2026/09/02]推薦・機械学習勉強会

### ISSUE THREAD

**Issue by ghibney** [wantedly/machine-learning-round-table#359]
## Why

推薦・機械学習勉強会は、推薦や機械学習、その周辺技術を通じてサービスを改善することにモチベーションのある人達の集まりです。ニュースやブログから論文まで、気になったものについてお互い共有しましょう！

発信のため、ここは **public** にしてあります。外部からの参加をご希望の方は[右手/ghibney](https://x.com/ghibney)、[坂元/___nsak](https://x.com/___nsak) まで DM を送るか、Wantedly Visit の募集（https://www.wantedly.com/projects/391912） よりご連絡ください！

## What

Wantedly では隔週水曜日に

- 推薦の評価指標について議論したい
- 〇〇っていうライブラリ / フレームワークを導入してみたい
- 他社の基盤事例をみんなにシェアして自社の基盤開発に活かしたい
- もっと推薦を良くするためにどんなものが必要か議論したい

といった話をする「推薦・機械学習勉強会」を開催しています。
この ISSUE はその会で話すネタを共有するための場所です。

話したいことがある人はここにコメントしましょう！
会の間に話した内容もここにメモしましょう！

prev: #356

**Comment by wakama1994** [comment]
## [ 機械学習基盤にFeature Store（Feast）をセルフホストで導入した](https://techblog.goinc.jp/entry/2026/08/31/161701)

FYI
Feastについて調べたら、参考記事でいかが出てきました。（Nogawanogawa-sanの記事）
https://www.nogawanogawa.com/entry/feast_intro

**Comment by chimuichimu** [comment]
## [推薦APIのモデル品質劣化を検知する監視システムの構築と運用](https://techblog.zozo.com/entry/recommend-api-model-monitoring)

**Comment by ghibney** [comment]
## [直前の検索クエリを考慮したYahoo!ショッピングの検索サジェスト改善](https://techblog.lycorp.co.jp/ja/20260818a)

### 概要
Yahoo!ショッピングの検索サジェストで、直前の検索クエリを入力に加えた BERT (Bidirectional Encoder Representations from Transformers) で候補をリランクし、オフライン評価と A/B テストの両方で改善を確認した。

### 詳細

* 入力文字列が同じでも検索意図は変わるため、集計値による並び替えでは候補が混在する

**同じ入力に複数の意図が混在する例**

<img width="1400" height="899" alt="Image" src="https://github.com/user-attachments/assets/da8e5084-7aba-48c6-a153-e1078e2e1cb4" />

* サジェストのクリックログでクリックされた候補を正例、同時に表示された候補を負例とする
* Yahoo!ショッピングの検索クエリで事前学習した BERT を `[CLS] 直前クエリ [SEP] 入力中クエリ [SEP] サジェスト候補 [SEP]` の形式でファインチューニングする
* 同じ表示機会の候補群のスコアを Softmax にかけ、クリックされた候補を正解として Cross Entropy loss を計算する

**学習データの作り方**

<img width="1400" height="957" alt="Image" src="https://github.com/user-attachments/assets/d9a5b033-24ec-486f-b79e-ff94ca07e385" />

**スコアリングと損失計算**

<img width="1400" height="572" alt="Image" src="https://github.com/user-attachments/assets/2f32abbb-f59c-4f89-8ff9-de3577d6cb6b" />

* オフライン評価は候補全件 (最大 20 件) のリランクで行い、A/B テストではサジェスト候補の CTR (Click Through Rate) が相対で約 7% 改善した

**オフライン評価の相対改善率 ※**
| 手法 | MRR (Mean Reciprocal Rank) | hits@1 | hits@3 |
|---|---|---|---|
| クリック数ベース | +2.4% | +9.3% | -1.4% |
| BERT (直前クエリなし) | +7.1% | +18.1% | +1.8% |
| BERT (直前クエリあり) | +11.6% | +26.9% | +5.2% |

※ 全て既存の表示順を基準とした相対改善率で、指標の絶対値ではない
※ MRR はクリックされた候補の順位の逆数の平均、hits@k はクリックされた候補が上位 k 件に入った割合

**「ワンピース」入力時のリランク結果**
| 順位 / 直前のクエリ | なし | レディース tシャツ | 呪術廻戦 フィギュア |
|---|---|---|---|
| 1 | ワンピース レディース | ワンピース レディース | ワンピースフィギュア |
| 2 | ワンピースカード | ワンピース 夏 | ワンピースカード |
| 3 | ワンピース 春 | ワンピース 春 | ワンピース レディース |
| 4 | ワンピース 夏 | ワンピース 長袖 | ワンピース 112巻 |
| 5 | ワンピース 長袖 | ワンピース 半袖 | ワンピース 夏 |

**「す」入力時のリランク結果**
| 順位 / 直前のクエリ | なし | スマホカバー | マリオカート | 靴 | ミネラルウォーター |
|---|---|---|---|---|---|
| 1 | スマートウォッチ | スマホケース | スイッチ2 | スニーカー | 水 |
| 2 | スニーカー | スニーカー | スマートウォッチ | スノーシューズ | スニーカー |
| 3 | スマホケース | スマートウォッチ | スニーカー | スマホケース | スマートウォッチ |
| 4 | 炊飯器 | スマホショルダー | スマホケース | スマートウォッチ | スマホケース |
| 5 | スーツケース | スマートリング | スマートタグ | スーツケース | 炊飯器 |
